app/exporters/shopify/product_exporter.rb

```diff
@@ -0,0 +1,90 @@
+module Shopify
+  class ProductExporter
+    def initialize(spree_product_id, factory = nil, logger = nil)
+      @logger = logger || default_logger
+      @factory = factory || default_factory
+      @spree_product = Spree::Product.find(spree_product_id)
+      @original_shopify_product = find_or_initialize(spree_product)
+    end
+
+    def perform
+      # NOTE(cab): Refactor the new.perform to use call?
+      shopify_product = factory.new(spree_product, original_shopify_product).perform
+      if shopify_product.save
+        save_pos_product_id(spree_product, shopify_product)
+        save_pos_variant_ids(spree_product, shopify_product.variants)
+        save_shopify_product_images(shopify_product, spree_product.variants)
+        logger.info("#{shopify_product.handle} imported")
+      else
+        logger.error("#{shopify_product.handle} not imported, reason: #{shopify_product.errors.full_messages}")
+      end
+
+      shopify_product
+    end
+
+    attr_accessor :logger, :factory, :spree_product, :original_shopify_product
+
+    private
+
+    def save_pos_product_id(product, shopify_product)
+      product.pos_product_id = shopify_product.id
+      product.save
+    end
+
+    def save_pos_variant_ids(spree_product, shopify_variants)
+      shopify_master_variant = shopify_variants.detect { |va| va.sku == spree_product.master.sku }
```

Comment:
This is dumb, instead of calling `spree_product.master_with_variants` I am treating the master variant differently. This could be majorly refactored.